sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -900,3 +901,35 @@ def test_precomputed_connectivity_affinity_with_2_connected_components():
 
     assert_array_equal(clusterer.labels_, clusterer_precomputed.labels_)
     assert_array_equal(clusterer.children_, clusterer_precomputed.children_)
+
+
+# TODO(1.4): Remove
+def test_deprecate_affinity():
+    rng = np.random.RandomState(42)
+    X = rng.randn(50, 10)
+    est = [
+        AgglomerativeClustering(),
+        AgglomerativeClustering(metric="euclidean"),
+    ]
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", FutureWarning)
+        for af in est:
+            af.fit(X)
```

Comment:
If those cases were already tested in the previous test, I would argue that this is enough. We have already turned `FutureWarning` into error for the full test suite.